### ISSUE TITLE
rename .dvc file to .lock file

### ISSUE THREAD

**Issue by skshetry** [iterative/dvc#4132]
We have two concepts now: `.dvc` file and `.lock` file. It'd be good to have consistent set of files: `dvc.lock` and `dvc.yaml`. The files could be named `.lock` (it could collide with other `.lock` files eg: Pipfile.lock, etc. so maybe `.dvc.lock`).

**Comment by efiop** [comment]
`.lock` is a temporary file that we can live without, which is not the case for `dvc add`ed and `dvc import`ed files right now. But even just when talking about renaming, `.dvc.lock` is the only option that is visually identifiable as something that belongs to dvc, at which point it is not clear why we would just make the prefix longer with no clear benefit.

**Comment by skshetry** [comment]
Having a single kind of format would be better to explain others conceptually. And, `dvc.lock` and `.dvc` files are the same thing with different file extension. Though, I did forget that it'd collide with other `.lock` files when making this issue.

**Comment by efiop** [comment]
@skshetry Actually they are not the same thing even anymore. E.g. `.lock` is multistage, but `.dvc` isn't. And the differences will be bigger and bigger as we go, as we are no longer limited by them (`dvc run` and `dvc add` .dvc-files) being the same thing. We used to have that in 0.x as you remember, and even though it was nice that it was the same format, but it was actually confusing in the real world, as data management and pipeline management didn't mix together nicely and we always had to explain all the corner cases (e.g. `dvc add` is a corner case of `dvc run` with no command, no deps), hence why we've moved to the current approach.
